### PR TITLE
Fix test_softmax, add test_log_softmax, fix Float generation

### PR DESCRIPTION
**Pull request by save-buffer**

### CODE REVIEW

**Comment by save-buffer** [comment]
Passes on both cosim and F1

File: regression/cuda/test_log_softmax.c [hunk 1]
```diff
@@ -0,0 +1,175 @@
+#include "test_log_softmax.h"
```
Comment (bornaehsani) [inline regression/cuda/test_log_softmax.c]: Could you copy the bsg_f1 license at the top of this file? You can find it at the top of all other files in bsg_f1.

File: regression/cuda/test_log_softmax.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#ifndef TEST_LOG_SOFTMAX_H
```
Comment (bornaehsani) [inline regression/cuda/test_log_softmax.h]: Could you copy the bsg_f1 license at the top of this file? You can find it at the top of all other files in bsg_f1.

**Comment by bornaehsani** [comment]
Other than the license the code looks great!

File: regression/cuda/test_log_softmax.c (hunk 1 above)
Comment (save-buffer) [reply regression/cuda/test_log_softmax.c]: Done!

File: regression/cuda/test_log_softmax.h (hunk 2 above)
Comment (save-buffer) [reply regression/cuda/test_log_softmax.h]: Done!

**Pull request merged**